src/transcoder.h

```diff
@@ -0,0 +1,17 @@
+#ifndef SRC_TRANSCODER_H_
+#define SRC_TRANSCODER_H_
+
+#include <stdlib.h>
+#include <stdint.h>
+
+namespace spellchecker {
+
+struct Transcoder;
+
+Transcoder *NewTranscoder();
+void FreeTranscoder(Transcoder *);
+bool TranscodeUTF16ToUTF8(const Transcoder *, char *out, size_t out_length, const uint16_t *in, size_t in_length);
```

Comment:
This class exists since the Hunspell spell-checker implementation needs to work on Windows as well as Linux. On Windows, it uses `WideCharToMultiByte`, and on Linux it uses `libiconv`.